RecoLocalMuon/GEMSegment/test/TestME0SegmentAnalyzer.cc

```diff
@@ -0,0 +1,301 @@
+#include "FWCore/ServiceRegistry/interface/Service.h"
+
+#include <DataFormats/GEMRecHit/interface/ME0SegmentCollection.h>
+ 
+#include "Geometry/GEMGeometry/interface/ME0Geometry.h"
+#include <Geometry/GEMGeometry/interface/ME0EtaPartition.h>
+#include <Geometry/Records/interface/MuonGeometryRecord.h>
+#include <DataFormats/MuonDetId/interface/ME0DetId.h>
+//
+// class declaration
+//
+
+class TestME0SegmentAnalyzer : public edm::EDAnalyzer {
+   public:
+      explicit TestME0SegmentAnalyzer(const edm::ParameterSet&);
+      ~TestME0SegmentAnalyzer();
+
+
+
+   private:
+      virtual void beginJob() ;
+      virtual void analyze(const edm::Event&, const edm::EventSetup&);
+      virtual void endJob() ;
+
+      virtual void beginRun(edm::Run const&, edm::EventSetup const&);
+      //virtual void endRun(edm::Run const&, edm::EventSetup const&);
+      //virtual void beginLuminosityBlock(edm::LuminosityBlock const&, edm::EventSetup const&);
+      //virtual void endLuminosityBlock(edm::LuminosityBlock const&, edm::EventSetup const&);
+
+      // ----------member data ---------------------------
+  edm::ESHandle<ME0Geometry> me0Geom;
+
+  edm::EDGetTokenT<ME0SegmentCollection> ME0Segment_Token;
+
+  std::string rootFileName;
+  std::unique_ptr<TFile> outputfile;
+  std::unique_ptr<TH1F> ME0_fitchi2;
+  std::unique_ptr<TH1F> ME0_Residuals_x;
+  std::unique_ptr<TH1F> ME0_Residuals_l1_x;
+  std::unique_ptr<TH1F> ME0_Residuals_l2_x;
+  std::unique_ptr<TH1F> ME0_Residuals_l3_x;
+  std::unique_ptr<TH1F> ME0_Residuals_l4_x;
+  std::unique_ptr<TH1F> ME0_Residuals_l5_x;
+  std::unique_ptr<TH1F> ME0_Residuals_l6_x;
+  std::unique_ptr<TH1F> ME0_Pull_x;
+  std::unique_ptr<TH1F> ME0_Pull_l1_x;
+  std::unique_ptr<TH1F> ME0_Pull_l2_x;
+  std::unique_ptr<TH1F> ME0_Pull_l3_x;
+  std::unique_ptr<TH1F> ME0_Pull_l4_x;
+  std::unique_ptr<TH1F> ME0_Pull_l5_x;
+  std::unique_ptr<TH1F> ME0_Pull_l6_x;
+  std::unique_ptr<TH1F> ME0_Residuals_y;
+  std::unique_ptr<TH1F> ME0_Residuals_l1_y;
+  std::unique_ptr<TH1F> ME0_Residuals_l2_y;
+  std::unique_ptr<TH1F> ME0_Residuals_l3_y;
+  std::unique_ptr<TH1F> ME0_Residuals_l4_y;
+  std::unique_ptr<TH1F> ME0_Residuals_l5_y;
+  std::unique_ptr<TH1F> ME0_Residuals_l6_y;
+  std::unique_ptr<TH1F> ME0_Pull_y;
+  std::unique_ptr<TH1F> ME0_Pull_l1_y;
+  std::unique_ptr<TH1F> ME0_Pull_l2_y;
+  std::unique_ptr<TH1F> ME0_Pull_l3_y;
+  std::unique_ptr<TH1F> ME0_Pull_l4_y;
+  std::unique_ptr<TH1F> ME0_Pull_l5_y;
+  std::unique_ptr<TH1F> ME0_Pull_l6_y;
+};
+
+//
+// constants, enums and typedefs
+//
+// constructors and destructor
+//
+TestME0SegmentAnalyzer::TestME0SegmentAnalyzer(const edm::ParameterSet& iConfig)
+
+{
+   //now do what ever initialization is needed
+  ME0Segment_Token = consumes<ME0SegmentCollection>(edm::InputTag("me0Segments"));
+
+  rootFileName  = iConfig.getUntrackedParameter<std::string>("RootFileName");
+  outputfile.reset(TFile::Open(rootFileName.c_str(), "RECREATE"));
```

Comment:
because it is there just for testing purposes. no need for running together with other analyzers, I ll consider this comment as minor.